src/core/derivation.ts

```diff
@@ -119,23 +119,24 @@ export function trackDerivedFunction<T>(derivation: IDerivation, f: () => T) {
 	derivation.runId = ++globalState.runId;
 	const prevTracking = globalState.trackingDerivation;
 	globalState.trackingDerivation = derivation;
-	let hasException = true;
+	let hasException = false;
 	let result: T;
 	try {
 		result = f.call(derivation);
-		hasException = false;
+	} catch(e){
+		hasException = true;
+		handleExceptionInDerivation(derivation, e);
+		throw e;
```

Comment:
The problem with this change is this changes the stack trace of the exception to this line. MobX had something similar in the past, but since the original stack is lost with a `catch`, people will blame MobX for the exception, which results in a plethorea of issues in the MobX issue tracker :). Sadly this cannot be fixed really nicely in Javascript as far as I know :cry: